components/treelist/state.md

```diff
@@ -4,9 +4,875 @@ page_title: TreeList - State
+                    items[i].HasChildren = false;
+                }
+            }
+        }
+    }
+
+
+    // sample model
+
+    public class Employee
+    {
+        public int Id { get; set; }
+
+        public string Name { get; set; }
+        public string EmailAddress { get; set; }
+        public DateTime HireDate { get; set; }
+
+        public List<Employee> DirectReports { get; set; }
+        public bool HasChildren { get; set; }
+
+        public override bool Equals(object obj)
+        {
+            if (obj is Employee)
+            {
+                return this.Id == (obj as Employee).Id;
+            }
+            return false;
+        }
+
+        public Employee()
+        {
+
+        }
+
+        public Employee(Employee itmToClone)
+        {
+            this.Id = itmToClone.Id;
+            this.Name = itmToClone.Name;
+        }
+
+        public static Employee GetClonedInstance(Employee itmToClone)
+        {
+            return new Employee(itmToClone);
+        }
+    }
+
+    // data generation
+
+    protected override async Task OnInitializedAsync()
+    {
+        Data = await GetTreeListData();
+    }
+
+    async Task<List<Employee>> GetTreeListData()
+    {
+        List<Employee> data = new List<Employee>();
+
+        for (int i = 1; i < 15; i++)
+        {
+            Employee root = new Employee
+            {
+                Id = LastId,
+                Name = $"root: {i}",
+                EmailAddress = $"{i}@example.com",
+                HireDate = DateTime.Now.AddYears(-i),
+                DirectReports = new List<Employee>(),
+                HasChildren = true
+            };
+            data.Add(root);
+            LastId++;
+
+            for (int j = 1; j < 4; j++)
+            {
+                int currId = LastId;
+                Employee firstLevelChild = new Employee
+                {
+                    Id = currId,
+                    Name = $"first level child {j} of {i}",
+                    EmailAddress = $"{currId}@example.com",
+                    HireDate = DateTime.Now.AddDays(-currId),
+                    DirectReports = new List<Employee>(),
+                    HasChildren = true
+                };
+                root.DirectReports.Add(firstLevelChild);
+                LastId++;
+
+                for (int k = 1; k < 3; k++)
+                {
+                    int nestedId = LastId;
+                    firstLevelChild.DirectReports.Add(new Employee
+                    {
+                        Id = LastId,
+                        Name = $"second level child {k} of {j} and {i}",
+                        EmailAddress = $"{nestedId}@example.com",
+                        HireDate = DateTime.Now.AddMinutes(-nestedId)
+                    }); ;
+                    LastId++;
+                }
+            }
+        }
+
+        data[0].Name += " (non-editable, see OnEdit)";
+
+        return await Task.FromResult(data);
+    }
+}
+````
+
```

Comment:
If time allows, add a section like this one https://docs.telerik.com/blazor-ui/components/grid/state#get-current-columns-visibility-order-field
If not, log a task